gobblin-service/src/main/java/org/apache/gobblin/service/modules/flow/BaseFlowToJobSpecCompiler.java

```diff
@@ -181,6 +190,18 @@ private  AddSpecResponse onAddFlowSpec(FlowSpec flowSpec) {
 
     // always try to compile the flow to verify if it is compilable
     Dag<JobExecutionPlan> dag = this.compileFlow(flowSpec);
+
+    if (this.warmStandbyEnabled &&
+        (!flowSpec.getConfigAsProperties().containsKey(ConfigurationKeys.JOB_SCHEDULE_KEY) || PropertiesUtils.getPropAsBoolean(flowSpec.getConfigAsProperties(), ConfigurationKeys.FLOW_RUN_IMMEDIATELY, "false"))) {
+      try {
+        // todo : we should probably check quota for all of the start nodes
+        userQuotaManager.checkQuota(dag.getNodes().get(0));
```

Comment:
we should iterate over and check all startNodes like we discussed right. Only if we have capacity for all start nodes, we add this dag.